### PR TITLE
Hind_Encorp all done

### PR DESCRIPTION
**Pull request by rahul-art**

### CODE REVIEW

**Comment by rahul-art** [comment]
ALL TEST PASSED locally

**Comment by skyprince999** [comment]
@rahul-art  kindly run the following from the datasets folder 

```
make style 
flake8 datasets

```

**Comment by rahul-art** [comment]
@skyprince999 I did that before it says all done

**Comment by rahul-art** [comment]
I did that again it gives the same output all done and then  I synchronized my changes with this branch

File: datasets/hind_encorp/README.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+---
+YAML tags:
+- copy-paste the tags obtained with the tagging app: http://34.68.228.168:8501/
```
Comment (lhoestq) [inline datasets/hind_encorp/README.md:3]: Can you add the YAML tags here ?
You can find more info at https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#tag-the-dataset-and-write-the-dataset-card

File: datasets/hind_encorp/hind_encorp.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add a description here."""
+
+from __future__ import absolute_import, division, print_function
+import datasets
```
Comment (lhoestq) [inline datasets/hind_encorp/hind_encorp.py:18]:
```suggestion

import datasets
```

File: datasets/hind_encorp/hind_encorp.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+                    Hindi.append(splits[4])
+            # last exampleguid
+            yield id_, {
+                "id": str(id_),
+                "source": source,
+                "alignment_type": alignment_type,
+                "alignment_quality": alignment_quality,
+                "English": English,
+                "Hindi": Hindi,
+            }
```
Comment (lhoestq) [inline datasets/hind_encorp/hind_encorp.py:173]: This looks wrong unfortunately
The data is composed of one pair of sentences per line, this is not the conll format.

Here is a suggestion of what it should look like in my opinion:

```python
            for id_, line in enumerate(f):
                splits = line.strip().split("\t")
                yield id_, {
                    "id": str(id_),
                    "source": splits[0],
                    "alignment_type": splits[1],
                    "alignment_quality": splits[2],
                    "English": splits[3],
                    "Hindi": splits[4],
                }
```

And another suggestion if you use the `Translation` feature type:

```python
            for id_, line in enumerate(f):
                splits = line.strip().split("\t")
                yield id_, {
                    "id": str(id_),
                    "source": splits[0],
                    "alignment_type": splits[1],
                    "alignment_quality": splits[2],
                    "translation": {
                        "en": splits[3],
                        "hi": splits[4]
                    },
                }
```

File: datasets/hind_encorp/hind_encorp.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+        features = datasets.Features(
+            {
+                "id": datasets.Value("string"),
+                "source": datasets.Sequence(datasets.Value("string")),
+                "alignment_type": datasets.Sequence(datasets.Value("string")),
+                "alignment_quality": datasets.Sequence(datasets.Value("string")),
+                "English": datasets.Sequence(datasets.Value("string")),
+                "Hindi": datasets.Sequence(datasets.Value("string"))
+                # These are the features of your dataset like images, labels ...
```
Comment (lhoestq) [inline datasets/hind_encorp/hind_encorp.py:86]: Can you use the `Translation` feature type ?

```suggestion
                "translation": datasets.features.Translation(languages=["en", "hi"])
```

File: datasets/hind_encorp/hind_encorp.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    def _info(self):
+        # TODO: This method pecifies the datasets.DatasetInfo object which contains informations and typings for the dataset
+        # This is the name of the configuration selected in BUILDER_CONFIGS above
+
+        features = datasets.Features(
+            {
+                "id": datasets.Value("string"),
+                "source": datasets.Sequence(datasets.Value("string")),
+                "alignment_type": datasets.Sequence(datasets.Value("string")),
+                "alignment_quality": datasets.Sequence(datasets.Value("string")),
```
Comment (lhoestq) [inline datasets/hind_encorp/hind_encorp.py:83]: If you apply my suggestion to parse one pair of sentences per line these should become
```suggestion
                "source": datasets.Value("string"),
                "alignment_type": datasets.Value("string"),
                "alignment_quality": datasets.Value("string"),
```

**Review by lhoestq — COMMENTED** [review]
Thanks ! good job with the dummy data.

I left a few comments

You'll need to regenerate the dataaset_infos.json file after these changes

**Comment by rahul-art** [comment]
@lhoestq i did all the changes you suggested but at the time of load_dataset it is giving me error
`**`datasets.utils.info_utils.NonMatchingSplitsSizesError: [{'expected': SplitInfo(name='train', num_bytes=76591256, num_examples=1, dataset_name='hind_encorp'), 'recorded': SplitInfo(name='train', num_bytes=78945714, num_examples=273885, dataset_name='hind_encorp')}]`**`

**Comment by skyprince999** [comment]
I cloned the branch and it seems to work fine at my end. try to clear the cache - 

```
rm -rf  /home/ubuntu/.cache/huggingface/datasets/
rm -rf  /home/ubuntu/.cache/huggingface/modules/datasets_modules//datasets/
```
But the dataset has only one record. Is that correct? 
image

**Comment by lhoestq** [comment]
> @lhoestq i did all the changes you suggested but at the time of load_dataset it is giving me error
> `**`datasets.utils.info_utils.NonMatchingSplitsSizesError: [{'expected': SplitInfo(name='train', num_bytes=76591256, num_examples=1, dataset_name='hind_encorp'), 'recorded': SplitInfo(name='train', num_bytes=78945714, num_examples=273885, dataset_name='hind_encorp')}]`**`

You can ignore this error by adding `ignore_verifications=True` to `load_dataset`.

This error is raised because you're loading a dataset that you've already loaded once in the past. Therefore the library does some verifications to make sure it's generated the same way. 

However since you've done changes in the dataset script you should ignore these verifications.

You can regenerate the dataset_infos.json with
```
datasets-cli test ./datasets/hindi_encorp --save_infos --all_configs --ignore_verifications
```


> I cloned the branch and it seems to work fine at my end. try to clear the cache -
> 
> ```
> rm -rf  /home/ubuntu/.cache/huggingface/datasets/
> rm -rf  /home/ubuntu/.cache/huggingface/modules/datasets_modules//datasets/
> ```
> 
> But the dataset has only one record. Is that correct?
> image

Yes the current parsing is wrong, I've already given @rahul-art some suggestions and it looks like it works way better now (num_examples=273885).

Thanks for fixing the parsing @rahul-art !
Feel free to commit and push your changes once it's ready :)

**Comment by rahul-art** [comment]
i ran the command you provided datasets-cli test ./datasets/hindi_encorp --save_infos --all_configs --ignore_verifications 
but now its giving this error

FileNotFoundError: Couldn't find file locally at ./datasets/hindi_encorp/hindi_encorp.py, or remotely at https://raw.githubusercontent.com/huggingface/datasets/master/datasets/./datasets/hindi_encorp/hindi_encorp.py or https://s3.amazonaws.com/datasets.huggingface.co/datasets/datasets/./datasets/hindi_encorp/hindi_encorp.py.
If the dataset was added recently, you may need to to pass script_version="master" to find the loading script on the master branch.

**Comment by lhoestq** [comment]
whoops I meant `hind_encorp` instead of `hindi_encorp` sorry

**Comment by rahul-art** [comment]
#1584 all changes have done successfully

**Comment by lhoestq** [comment]
Ok thanks ! closing this one in favor of #1584

**Pull request closed without merging**